### ISSUE TITLE
False positive for no-alert

### ISSUE THREAD

**Issue by ilyavolodin** [eslint/eslint#2241]
I'm not 100% sure this should be fixed but the following code:
```js
var alert = function(message) {
   console.log(message);
};
alert('Hello');
```
Will show `Unexpected alert.` message. However, in this case alert is being overridden (if you turned off no-native-reassign). Same thing happens with `no-console` rule.
Theoretically, `no-alert` should check if `alert` and `confirm` has been overridden, before warning about it.

**Comment by nzakas** [comment]
We purposely kept this simple so we didn't have to keep track of shadows. If you want to try making a change, it's fine by me, I'm just not sure how valuable it is.

**Comment by ilyavolodin** [comment]
Yes, same thing here. I'm not positive this is needed. I'll wait to see if anyone else feels strongly about it.

**Comment by lo1tuma** [comment]
Looks like a duplicate of #1105

**Comment by ilyavolodin** [comment]
Yes, good call. Duplicate of #1105
